easybuild/easyblocks/c/cmake.py

```diff
@@ -0,0 +1,118 @@
+##
+# Copyright 2020 Alexander Grund
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+@author: Alexander Grund
+"""
+import os
+
+from easybuild.easyblocks.generic.configuremake import ConfigureMake
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.config import build_option
+from easybuild.tools.modules import get_software_root, get_software_libdir
+import easybuild.tools.environment as env
+
+
+class EB_CMake(ConfigureMake):
+    """
+    EasyBlock to install CMake
+    """
+
+    @staticmethod
+    def extra_options():
+        extra_vars = ConfigureMake.extra_options()
+        extra_vars.update({
+            'use_openssl': [True, "Use openSSL (if building CURL in CMake)", CUSTOM],
+        })
+        return extra_vars
+
+    def configure_step(self):
+        """
+        Run qmake on the GUI, if necessary
+        """
+        configopts = self.cfg['configopts'].split(' -- ', 1)
+        configure_opts = configopts[0]
+        cmake_opts = configopts[1] if len(configopts) == 2 else ''
+
+        add_configure_opts = {'parallel': '%(parallel)s'}
```

Comment:
This should probably be something like
add_configure_opts = {}
if self.cfg['parallel']:
  add_configure_opts['parallel'] = self.cfg['parallel']

Reply:
Really? In the CMake ECs the unconditional '%(parallel)s' works well and IMO `parallel` is always set by EB

Reply:
The default for parallel in ./easybuild/framework/easyconfig/default.py is None but should probably always be set but...

Reply:
It will be set as "None" means "autodetect": https://github.com/easybuilders/easybuild-framework/blob/090f8c1d0cf0837eda084abd6f7ba53aaa476d7e/easybuild/framework/easyblock.py#L1618-L1633